### ISSUE TITLE
Numbers at end of strings aren't highlighted as part of the string

### ISSUE THREAD

**Issue by Caged** [atom/language-sql#30]
I updated to 1.21 and noticed this started to happen.  The digits at the end of the string should be the same color as the string.

![screen shot 2015-11-18 at 10 57 17 am](https://cloud.githubusercontent.com/assets/25/11250806/2dd8dd0a-8de3-11e5-9cf6-4152552243c9.png)

```
language-sql
├── 0.19.0
├── https://github.com/atom/language-sql
├── SQL language support in Atom
├── 120180 downloads
└── 29 stars
```

**Comment by DavidLGoldberg** [comment]
Yeah, this is definitely new in 1.21.  I'm writing a semantic highlighting package, but utilizing the grammar, and a few of my tests broke :(

Is this going to be fixed soon.  I'd rather not have to work around it :-)

**Comment by 50Wliu** [comment]
I'll see what I can do today.

**Comment by DavidLGoldberg** [comment]
@50Wliu, I'd love to take a look tonight and this weekend, but couldn't quickly spot the commit, do you happen to know which one it was that broke it?

**Comment by 50Wliu** [comment]
@DavidLGoldberg If I had to guess, #24 (there was a small modification in 05efcd8e8594c083893da17c8b50f4d3847502a0 though).  I think an underscore is a valid word boundary so you're going to have to work around that through a negative lookbehind.

**Comment by DavidLGoldberg** [comment]
The SHA above is definitely the one that breaks it from what I can tell with:
`apm develop language-sql` and restoring the previous.

It also passes all of my tests (I thought I introduced another issue, but it turns out it was just that line from the SHA :\ )

I'll try to investigate further, as far as seeing if it potentially breaks something else too?  Although, I have to confess I have very little chance of helping out with that regex :(
Where are the best docs/tutorials on this grammar stuff btw?  Sublime community still maybe?

**Comment by 50Wliu** [comment]
Docs are somewhat lacking on grammar development, unfortunately.  In general Atom follows the [Textmate guidelines](https://manual.macromates.com/en/language_grammars) since most of the grammars were converted from Textmate, but the guidelines are pretty vague.
There's an ongoing PR to develop a more cohesive standard in atom/docs#114.

**Comment by diesire** [comment]
I'm having the same problem with `v0.20.0`

![image](https://cloud.githubusercontent.com/assets/1185591/11809750/5347bcfe-a329-11e5-9570-4326f043b016.png)

Note that table name `T1` is OK, but all fields are broken

![image](https://cloud.githubusercontent.com/assets/1185591/11810033/60493624-a32b-11e5-9aa5-756fbd671d77.png)

I guess it's because `T1` is recognized as an entity but `FIELD` not, so in the first the entity takes precedence over the number but in the last the number rule applies

**Comment by 50Wliu** [comment]
Fixed by #34.
